src/webgpu/gpu_test.ts

```diff
@@ -162,6 +163,10 @@ class DevicePool {
     }
     if (gpuOutOfMemoryError !== null) {
       assert(gpuOutOfMemoryError instanceof GPUOutOfMemoryError);
+
+      // Try to clean up, in case there are stray GPU resources in need of collection.
+      await attemptGarbageCollection();
```

Comment:
nit: a future improvement may be to do this _after_ setting `this.holder` to be undefined in `release(GPUDevice)` so that the GPUDevice is unreachable. More complexity involved with bubbling that signal out and hard to know how useful it will be, so a TODO is also okay.

Reply:
Not so hard to do, and if we don't do it now we may never do it. Take a look at the first new commit (second one is just a move)